node/network/approval-distribution/src/lib.rs

```diff
@@ -1706,22 +1799,35 @@ impl ApprovalDistribution {
 				},
 				FromOrchestra::Signal(OverseerSignal::Conclude) => return,
 			}
+
+			if maybe_send_reputation(
+				ctx.sender(),
+				reputation,
+				&mut reputation_delay,
+				&reputation_interval,
+			)
+			.await
+			{
+				reputation.clear();
+				reputation_delay = futures_timer::Delay::new(reputation_interval).fuse()
+			}
 		}
 	}
 
 	async fn handle_incoming<Context>(
 		ctx: &mut Context,
 		state: &mut State,
+		reputation: &mut ReputationAggregator,
```

Comment:
I'd have the `ReputationAggregator` part of the `State` to avoid an extra argument to the function.

Reply:
Maybe we even don't need to use extra abstraction and can just hold the hashmap with aggregated reputations in the State?